packages/features/insights/server/trpc-router.ts

```diff
@@ -1534,7 +1450,22 @@ export const insightsRouter = router({
       return result;
     }),
   rawData: userBelongsToTeamProcedure.input(rawDataInputSchema).query(async ({ ctx, input }) => {
-    const { startDate, endDate, teamId, userId, memberUserId, isAll, eventTypeId } = input;
+    const { startDate, endDate, teamId, userId, memberUserId, isAll } = input;
+
+    const eventTypeList = await getEventTypeList({
+      prisma: ctx.prisma,
+      teamId,
+      userId,
+      isAll,
+      user: {
+        id: ctx.user.id,
+        organizationId: ctx.user.organizationId,
+        isOwnerAdminOfParentTeam: ctx.user.isOwnerAdminOfParentTeam,
+      },
+    });
+
+    // use eventTypeId filter only if it's accessible by this user
+    const eventTypeId = eventTypeList.find((eventType) => eventType.id === input.eventTypeId)?.id;
```

Comment:
checking eventTypeId in case the frontend is sending eventTypeId that this user isn't supposed to access.